spectacles/validators.py

```diff
@@ -194,9 +195,25 @@ def build_project(self, selectors: List[str]) -> None:
         selected_models = self._select(
             choices=tuple(selection.keys()), select_from=project_models
         )
+        excluded_models = self._select(
+            choices=tuple(exclusion.keys()), select_from=project_models
+        )
 
-        for model in selected_models:
+        excluded_explores = {}
+        for model in excluded_models:
             # Expand wildcard operator to include all specified or discovered explores
+            excluded_explore_names = exclusion[model.name]
+            if "*" in excluded_explore_names:
+                excluded_explore_names.remove("*")
+                excluded_explore_names.update(
+                    set(explore.name for explore in model.explores)
+                )
```

Comment:
If there's a situation with `model_A.explore_A` and `model_B.explore_A`, and the `--exclude` argument is `model_A.explore_A`, would this still preserve `explore_A` in `model_B`? Any way we can add this case to the unit tests?

Reply:
It will still preserve it. The explores to exclude are stored in lists against a key of the model name. We then only check against the explores within a given model: https://github.com/spectacles-ci/spectacles/pull/172/files#diff-cbee2ab0268f8f001868af4ca066ba04R228-R232

I'll definitely add some tests to confirm that though.

Reply:
@joshtemple I've confirmed this is the case and have added an additional test `test_build_project_one_ambiguous_explore_excluded` to verify so.

I also actually re-factored some of the other tests to make the selection easier.